staging/src/k8s.io/client-go/util/workqueue/delaying_queue_test.go

```diff
@@ -174,6 +174,44 @@ func TestAddTwoFireEarly(t *testing.T) {
 	}
 }
 
+func TestAddAt(t *testing.T) {
+	now := time.Now()
+	fakeClock := testingclock.NewFakeClock(now)
+	q := NewDelayingQueueWithCustomClock(fakeClock, "")
+
+	addedAtPast := "addedAtPast"
+	addedAtNow := "addedAtNow"
+	addedAtFuture := "addedAtFuture"
+
+	q.AddAt(addedAtPast, now.Add(-50*time.Millisecond))
+	q.AddAt(addedAtNow, now)
+	q.AddAt(addedAtFuture, now.Add(50*time.Millisecond))
+
+	if err := waitForWaitingQueueToFill(q); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	if err := waitForAdded(q, 2); err != nil {
```

Comment:
we shouldn't need to wait... the first two elements should be added synchronously, IIUC.

Reply:
Yes, done